docker/src/main/java/org/arquillian/cube/impl/client/CubeSuiteLifecycleController.java

```diff
@@ -100,22 +100,28 @@ private boolean isCubeRunning(String cube) {
     }
 
     private final class StartCubes implements Callable<RuntimeException> {
-        private final boolean allowToConnectToRunningContainers;
+        private final ConnectionMode connectionMode;
         private final String cubeId;
 
-        private StartCubes(String cubeId, boolean shouldAllowToConnectToRunningContainers) {
+        private StartCubes(String cubeId, ConnectionMode connectionMode) {
             this.cubeId = cubeId;
-            this.allowToConnectToRunningContainers = shouldAllowToConnectToRunningContainers;
+            this.connectionMode = connectionMode;
         }
 
         @Override
         public RuntimeException call() throws Exception {
             try {
-                if(allowToConnectToRunningContainers && isCubeRunning(cubeId)) {
+                if(connectionMode.isAllowReconnect() && isCubeRunning(cubeId)) {
                     controlEvent.fire(new PreRunningCube(cubeId));
-                } else {
-                    controlEvent.fire(new CreateCube(cubeId));
-                    controlEvent.fire(new StartCube(cubeId));
+                    return null;
+                }
+                controlEvent.fire(new CreateCube(cubeId));
+                controlEvent.fire(new StartCube(cubeId));
+
+                if(connectionMode.isAllowReconnect() && !connectionMode.isStoppable()) {
+                 // If we allow reconnections and containers are none stoppable which means that they will be able to be
+                 // reused in next executions then at this point we can assume that the container is a prerunning container.
+                 controlEvent.fire(new PreRunningCube(cubeId));
```

Comment:
Hmm.. are we missing some else or something here? Why create/start then call PreRunningCube?